ocaml/idl/datamodel_certificate.ml

```diff
@@ -32,9 +39,15 @@ let t =
     ~messages_default_allowed_roles:_R_READ_ONLY
     ~contents:
       [ uid   _certificate ~lifecycle
-      ; field   ~qualifier:StaticRO ~lifecycle
+      ; field ~qualifier:DynamicRO ~lifecycle:[Published, rel_next, ""]
+          ~ty:certificate_type "type"
+          "The type of certificate, can be a host certificate, a CA certificate, or both"
```

Comment:
Do we need to assign a default value or otherwise handle upgrades?

Reply:
Upgrades need to be handled, all existing certificates are have the type `` `host``.
Similarly, the validated_by need to be populated.

Reply:
If you synchronise all `Certificate` objects for a host in the dbsync phase of xapi's startup sequence, then you don't need special "upgrade" code.

Reply:
I've dropped the `db_upgrade` code, the new fields will be updated in `update_env` from `dbsync_master.ml` instead. I'd like to do that in a separate PR and merge this to build an SDk with these changes.

Reply:
I assume the FQP will contain some test that checks that upgrades work when you got these certificates installed (wouldn't rely on RPU tests catching this, they don't use the full features of the product).